pkg/skaffold/schema/transform.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schema
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/skaffold/pkg/skaffold/config"
+	"github.com/GoogleCloudPlatform/skaffold/pkg/skaffold/schema/util"
+	"github.com/pkg/errors"
+)
+
+type Transform func(vc *util.VersionedConfig) (*util.VersionedConfig, error)
+
+// Map of schema version to transform functions
+// If there are n schema versions, this should always contain (n-1) entries,
+// since the last schema version should not have a transform
+var transformers map[string]Transform = map[string]Transform{}
+
+func RunTransform(vc *util.VersionedConfig) (*util.VersionedConfig, error) {
```

Comment:
can you write some tests for this?

Reply:
yeah definitely, might make sense to add them in a follow up PR once we make a config change though? since we don't have any other schemas it's a little hard to test right now.